### ISSUE TITLE
[CFE]: Fix horizontal scrolling in Analytics tables

### ISSUE THREAD

**Issue by tomalec** [woothemes/woocommerce#53017]
### Which release does this request apply to?

9.5

### Which PR needs to be included? (please do not enter multiple PRs)

https://github.com/woocommerce/woocommerce/pull/52971

### Why do these PRs need to break the code freeze?

WC 9.2.0 [Introduced a change](https://github.com/woocommerce/woocommerce/pull/48487) that effectively broke horizontal scroll in Analytics tables (which makes them quite not accessible at all on mobile) 
see the issue https://github.com/woocommerce/woocommerce/issues/52879
(discussion with context p1731625358851769-slack-C0410KV3YLW)

### What is the consequence if this exception does not get included?

The tables in Analytics will not be usable by users with narrower viewports, like mobile.

### What is the plan should defects to be discovered in these PRs ahead of final release?

The PR could be reverted. It's a single line in CSS, so merge conflicts are unlikely and easy to solve.

### How should this change be communicated in the release post on the public developer blog:

I think there is no need for additional communication. It's a bug fix that should not disturb or affect existing workflows.

### If you’re not available and we have questions about this request, is there another person(s) and/or a team that we can ping?

@eason9487 or @mikkamp 

### Acknowledgement

- [x] I understand that I need to write an incident report (aka Escape Analysis) as a comment on this post. This is required for the request to be accepted.
- [x] I understand that I need to create an issue as a result of the Escape Analysis and reference it in a comment on this post. This can be done at a later time, but it is required for this request to be closed.

**Comment by tammullen** [comment]
- [x] Escape analysis completed
- [x] PR merged to target release branch
- [x] PR cherry-picked to trunk
- [ ] Followup issue if any

**Comment by tammullen** [comment]
@tomalec Reminder to do an escape analysis for this

**Comment by tomalec** [comment]
### Escape analysis

#### Investigation
Based on our Team's investigation (p1731625358851769-slack-C0410KV3YLW), the mobile user experience of Analytics was broken due to a few seemingly unrelated factors. That's probably why it went under the radar.
The fix for banner widths (https://github.com/woocommerce/woocommerce/pull/48487) made `home screen/style.scss` be loaded on the Analytics page.
That stylesheet included quite an intrusive `!important` rule originally introduced in https://github.com/woocommerce/woocommerce-admin/pull/4347, 

```css
.woocommerce-admin-page #wpcontent,
.woocommerce-admin-page.woocommerce_page_wc-admin #wpbody-content {
	overflow-x: inherit !important; // Overwriting wc-admin css from elsewhere. Necessary to make the right-hand column 'stick'
	position: relative;
}
```
That, in this case, is resolved to `overflow-x: hidden` - which is precisely the way to block horizontal scroll.

#### The fix

We prepared [a quick fix](https://github.com/woocommerce/woocommerce/pull/52971) by narrowing down the selector for this rule to extensions and home screen pages only so the horizontal scrolling won't be blocked on unrelated pages. 
```css
.woocommerce-admin-page__home #wpcontent,
.woocommerce-admin-page__home.woocommerce_page_wc-admin #wpbody-content,
.woocommerce-admin-page__extensions #wpcontent,
.woocommerce-admin-page__extensions.woocommerce_page_wc-admin #wpbody-content {
```
However, it's worth revisiting that fix to check if more pages should share this behavior or if we could implement sticky panels in some other ways. (cc @psealock)

#### How could we have avoided this?

It can be challenging to check all possible scenarios and side effects of CSS rules applied to the same light DOM and all the pages. Also, preparing automated tests for all looks and behavior could be difficult to achieve in both a complete and efficient manner.

Maybe @woocommerce/ghidorah  could add something more about how we could have prevented that bug from happening. Not having the full context of those PRs, I can only serve some general ideas:
 
- Test the PRs on mobile views/narrower views and with more complete store data  - I've noticed the Customers page (which technically is the Analytics page) was checked during the review (https://github.com/woocommerce/woocommerce/pull/48487#pullrequestreview-2140446738), so if we have had it populated with customers data and rendered in a smaller window, the chance of spotting the bug would be slightly higher. (However, still, one would have to touch and scroll a lot)
 - Take greater care while scoping the CSS selectors, especially those with `! important` statements and behavior-blocking effects.

#### Why did it have to break the CF?

Even though the issue was present before the release. It was prioritized to be shipped with the frozen release, as our leadership spotted the issued complaint on X (https://x.com/tombrunberg/status/1857166309559066902). Therefore, we wanted to bring back the mobile experience to Analytics ASAP to avoid bad publicity.

**Comment by ilyasfoo** [comment]
Thanks for fixing this issue, @tomalec! 

> Take greater care while scoping the CSS selectors, especially those with ! important statements and behavior-blocking effects.

I agree, I think we're doing well with not having too much of these around. If necessary, we could go through the codebase and refactor some of the existing ones to reduce tech debt, but I'm unsure if it's worth the effort.
